lib/src/builders/message/component.dart

```diff
@@ -54,31 +54,31 @@ class ButtonBuilder extends MessageComponentBuilder {
   ButtonBuilder.primary({
     this.label,
     this.emoji,
-    required String customId,
+    required this.customId,
```

Comment:
```suggestion
    required String this.customId,
```

`this.customId` is nullable (as URL buttons do not have a custom ID), but it must be provided in this constructor. The bug here was just a missing `this.` before `customId`.